### ISSUE TITLE
Deno gfm vendoring issues

### ISSUE THREAD

**Issue by mcandeia** [denoland/deno#21415]
Version: Deno 1.38.4

I encountered a bug related to vendoring dependencies while using deno. The problem becomes apparent when attempting to vendor the [deno-gfm](https://github.com/denoland/deno-gfm) library. For detailed information on how to reproduce this issue, please refer to the repository:

https://github.com/mcandeia/deno-vendor-bug

To reproduce, fork the repository and try to run with `--no-remote` options poiting to vendor/import_map.json file

**Comment by mcandeia** [comment]
@dsherret I couldn't make it work either, it's not clear to me if I should run `deno cache --vendor main.tsx` and then I can run with `--no-remote` ? previously It was possible by using the generated import_map inside vendor folder, but looks like it was removed, right?

**Comment by mcandeia** [comment]
ping @dsherret

**Comment by dsherret** [comment]
Upgrading to deno-gfm 0.3.0 should fix the issue now.

**Comment by mcandeia** [comment]
Thanks, will check @dsherret

**Comment by mcandeia** [comment]
Now it's a different problem,
Tested with: https://github.com/deco-sites/start

(using Deno 1.39.1)
`deno cache --vendor dev.ts` will generate a vendor folder with no import_map generated. I don't know if I did something wrong, but couldn't make it work @dsherret 

<img width="622" alt="image" src="https://github.com/denoland/deno/assets/5839364/968274eb-7898-45f6-9cc8-70d88e08fba4">

**Comment by dsherret** [comment]
Yeah, that's expected. The `--vendor` feature (recommended use as `{ "vendor": true }` in deno.json) is a completely different more reliable way of doing vendoring that does a cache override and doesn't use an import map (https://deno.com/blog/v1.37#vendor-as-cache-override-unstable). I wouldn't recommend using it yet though because it's not supported on Deploy yet.

Regarding the original `deno vendor` issue, it seems there's a new issue that's surfaced:

```
> deno vendor main.tsx
error: Failed resolving types. Relative import path "Instance" not prefixed with / or ./ or ../ and not in import map from "https://esm.sh/v135/marked@9.1.1/lib/marked.d.ts"
    at https://esm.sh/v135/marked@9.1.1/lib/marked.d.ts:553:39
```

**Comment by mcandeia** [comment]
> Yeah, that's expected. The --vendor feature (recommended use as { "vendor": true } in deno.json) is a completely different more reliable way of doing vendoring that does a cache override and doesn't use an import map

Oh, I forgot to mention that the problem is exactly that, it changed my deno.json file and points to an import_map.json that do not exists @dsherret.

**Comment by mcandeia** [comment]
Just to let you know. I changed my approach because I couldn't make it worth with vendor command nor --vendor option. It was much simpler to just zip the entire `DENO_DIR` and share it between machines. It seems to be a poor-man's solution, but I think vendoring is not a deno priority right now.

**Comment by nayeemrmn** [comment]
It seems the follow-up issue had to do with an old import map from `deno vendor`.
